custom/motioncontrollers/MotionController.java

```diff
@@ -63,14 +89,33 @@ public MotionController(PIDSource source) {
 	 *
 	 * @warning this does not indicate sensor errors
 	 */
-	public abstract void reset();
+	public final void reset() {
+		resetMC();
+		setpoint = sensor.pidGet();
+		synchronized (justReset) {
+			justReset = true;
+		}
```

Comment:
Useless for two reasons: synchronizing on a field that changes doesn't do anything, and setting the value of something doesn't need to be synchronized (what matters for synchronizing is blocks of code that require the value to remain the same)